rust/main/chains/hyperlane-sealevel/src/transaction/tests.rs

```diff
@@ -1,29 +1,57 @@
+use solana_sdk::pubkey::Pubkey;
 use solana_transaction_status::EncodedTransactionWithStatusMeta;
 
-use crate::transaction::search_dispatched_message_transactions;
+use crate::transaction::{
+    is_message_delivery_instruction, is_message_dispatch_instruction, search_message_transactions,
+};
 use crate::utils::decode_pubkey;
 
 #[test]
 pub fn test_search_dispatched_message_transaction() {
     // given
-    let mailbox_program_id = decode_pubkey("E588QtVUvresuXq2KoNEwAmoifCzYGpRBdHByN9KQMbi").unwrap();
     let dispatched_message_pda_account =
         decode_pubkey("6eG8PheL41qLFFUtPjSYMtsp4aoAQsMgcsYwkGCB8kwT").unwrap();
-    let transaction = serde_json::from_str::<EncodedTransactionWithStatusMeta>(JSON).unwrap();
-    let transactions = vec![transaction];
+    let (mailbox_program_id, transactions) = transactions(DISPATCH_TXN_JSON);
 
     // when
-    let transaction_hashes = search_dispatched_message_transactions(
+    let transaction_hashes = search_message_transactions(
         &mailbox_program_id,
         &dispatched_message_pda_account,
         transactions,
+        &is_message_dispatch_instruction,
     );
 
     // then
     assert!(!transaction_hashes.is_empty());
 }
 
-const JSON: &str = r#"
+#[test]
+pub fn test_search_delivered_message_transaction() {
+    // given
+    let delivered_message_pda_account =
+        decode_pubkey("Dj7jk47KKXvw4nseNGdyHtNHtjPes2XSfByhF8xymrtS").unwrap();
+    let (mailbox_program_id, transactions) = transactions(DELIVERY_TXN_JSON);
+
+    // when
+    let transaction_hashes = search_message_transactions(
+        &mailbox_program_id,
+        &delivered_message_pda_account,
+        transactions,
+        &is_message_delivery_instruction,
+    );
+
+    // then
+    assert!(!transaction_hashes.is_empty());
+}
+
+fn transactions(json: &str) -> (Pubkey, Vec<EncodedTransactionWithStatusMeta>) {
+    let mailbox_program_id = decode_pubkey("E588QtVUvresuXq2KoNEwAmoifCzYGpRBdHByN9KQMbi").unwrap();
+    let transaction = serde_json::from_str::<EncodedTransactionWithStatusMeta>(json).unwrap();
+    let transactions = vec![transaction];
+    (mailbox_program_id, transactions)
+}
+
+const DISPATCH_TXN_JSON: &str = r#"
```

Comment:
nit: would be nice to eventually read these jsons from a file

Reply:
I'll implement it in next PR.